Invalid C code generated

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.2.2 ea0ac09
**OS:** macOS 10.15

**What did you do?**

```v
struct Nofun {
        foo fn (a int)
}

interface NofunInterface {
        foo fn (a int)
}

fn (nf Nofun)foo(a int) {
        println('Hello world')
}

fn main() {
        nf := NofunInterface(Nofun{})
        nf.foo(21)
}
```

**What did you expect to see?**

I think this code should be valid
 
**What did you see instead?**

invalid C code generated:
```
==================
/var/folders/12/517hk27n0pxbnbv9yxlh4jxm0000gn/T/v/nofun.4602538498403763089.tmp.c:9391:8: error: called object type 'anon_fn_int *' (aka 'void (**)(int)') is not a function or function pointer
        nf.foo(21);
        ~~~~~~^
1 error generated.
...
```


Note that removing the `fn` keyword in the interface definition makes the program compile and run fine:

```v
struct Nofun {
        foo fn (a int)
}

interface NofunInterface {
        foo (a int)
}

fn (nf Nofun)foo(a int) {
        println('Hello world')
}

fn main() {
        nf := NofunInterface(Nofun{})
        nf.foo(21)
}
```

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -757,7 +757,13 @@ fn (mut g Gen) fn_call(node ast.CallExpr) {
 	// TODO: test node.left instead
 	// left & left_type will be `x` and `x type` in `x.fieldfn()`
 	// will be `0` for `foo()`
+	mut is_interface_call := false
 	if node.left_type != 0 {
+		left_sym := g.table.get_type_symbol(node.left_type)
+		if left_sym.kind == .interface_ {
+			is_interface_call = true
+			g.write('(*')
+		}
 		g.expr(node.left)
 		if node.left_type.is_ptr() {
 			g.write('->')
@@ -889,8 +895,12 @@ fn (mut g Gen) fn_call(node ast.CallExpr) {
 			// g.writeln(';')
 			// g.write(cur_line + ' /* <== af cur line*/')
 			// }
+			g.write(g.get_ternary_name(name))
+			if is_interface_call {
+				g.write(')')
+			}
 			mut tmp_cnt_save := -1
-			g.write('${g.get_ternary_name(name)}(')
+			g.write('(')
 			if g.is_json_fn {
 				g.write(json_obj)
 			} else {

--- a/vlib/v/tests/interface_fields_test.v
+++ b/vlib/v/tests/interface_fields_test.v
@@ -54,3 +54,21 @@ fn test_interface_fields() {
 	assert c.breed == 'what??'
 	assert d.breed == 'what??'
 }
+
+struct Nofun {
+	foo fn (a int) int
+}
+
+interface NofunInterface {
+	foo fn (a int) int
+}
+
+fn my_fn(a int) int {
+	assert a == 123
+	return a * 2
+}
+
+fn test_interface_fn_pointer_fields() {
+	nf := NofunInterface(Nofun{ my_fn })
+	assert nf.foo(123) == 246
+}

--- a/vlib/v/tests/interface_fields_test.v
+++ b/vlib/v/tests/interface_fields_test.v
@@ -56,11 +56,11 @@ fn test_interface_fields() {
 }
 
 struct Nofun {
-	foo fn (a int) int
+	foo fn (int) int
 }
 
 interface NofunInterface {
-	foo fn (a int) int
+	foo fn (int) int
 }
 
 fn my_fn(a int) int {
@@ -69,6 +69,6 @@ fn my_fn(a int) int {
 }
 
 fn test_interface_fn_pointer_fields() {
-	nf := NofunInterface(Nofun{ my_fn })
+	nf := NofunInterface(Nofun{my_fn})
 	assert nf.foo(123) == 246
 }

